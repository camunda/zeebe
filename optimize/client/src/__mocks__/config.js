/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

export const isEmailEnabled = () => true;
export const isSharingEnabled = () => true;
export const getOptimizeVersion = () => '2.7.0';
export const getWebappEndpoints = () => ({
  default: {endpoint: 'http://localhost:8080/camunda', engineName: 'default'},
});
export const getHeader = () => ({
  textColor: 'dark', // or "light"
  backgroundColor: '#FFFFFF',
  logo: '',
});
export const ConfigProvider = () => <div />;
export {default as newReport} from '../modules/newReport.json';
