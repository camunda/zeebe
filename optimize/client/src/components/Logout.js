/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {useEffect} from 'react';
import {withRouter} from 'react-router-dom';
import {get} from 'request';
import {withErrorHandling} from 'HOC';
import {addNotification} from 'notifications';
import {t} from 'translation';

export function Logout({mightFail, history}) {
  useEffect(() => {
    (async () => {
      await mightFail(
        get('api/authentication/logout'),
        () => addNotification({text: t('navigation.logoutSuccess')}),
        () => addNotification({text: t('navigation.logoutFailed'), type: 'error'})
      );
      history.replace('/');
    })();
  }, [mightFail, history]);

  return null;
}

export default withRouter(withErrorHandling(Logout));
