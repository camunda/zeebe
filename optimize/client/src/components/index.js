/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

export {Analysis} from './Analysis';
export {Footer} from './Footer';
export {Header} from './Header';
export {PrivateRoute} from './PrivateRoute';
export {Dashboard} from './Dashboards';
export {Report} from './Reports';
export {Sharing} from './Sharing';
export {Home, Collection} from './Home';
export {default as Logout} from './Logout';
export {Processes, ProcessReport} from './Processes';
