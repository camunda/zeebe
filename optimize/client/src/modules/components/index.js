/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

export {Table} from './Table';
export {Button} from './Button';
export {Modal} from './Modal';
export {Dropdown} from './Dropdown';
export {Deleter} from './Deleter';
export {Input} from './Input';
export {Select} from './Select';
export {BPMNDiagram, ClickBehavior, PartHighlight} from './BPMNDiagram';
export {DMNDiagram} from './DMNDiagram';
export {CopyToClipboard} from './CopyToClipboard';
export {ReportRenderer, NoDataNotice} from './ReportRenderer';
export {ErrorBoundary} from './ErrorBoundary';
export {Icon} from './Icon';
export {ActionItem} from './ActionItem';
export {ShareEntity} from './ShareEntity';
export {DashboardRenderer, DiagramScrollLock} from './DashboardRenderer';
export {DefinitionSelection, VersionPopover, TenantPopover} from './DefinitionSelection';
export {TargetValueBadge} from './TargetValueBadge';
export {ErrorPage} from './ErrorPage';
export {Labeled} from './Labeled';
export {LabeledInput} from './LabeledInput';
export {DatePicker} from './DatePicker';
export {LoadingIndicator} from './LoadingIndicator';
export {ColorPicker} from './ColorPicker';
export {EntityName} from './EntityName';
export {EntityNameForm} from './EntityNameForm';
export {SubNav, NavItem} from './navigation';
export {HeatmapOverlay} from './HeatmapOverlay';
export {EntityList} from './EntityList';
export {UserTypeahead} from './UserTypeahead';
export {DateRangeInput} from './DateRangeInput';
export {LastModifiedInfo} from './LastModifiedInfo';
export {ReportDetails} from './ReportDetails';
export {InstanceCount} from './InstanceCount';
export {DocsLink} from './DocsLink';
export {Tabs} from './Tabs';
export {Checklist} from './Checklist';
export {Tooltip} from './Tooltip';
export {SelectionPreview} from './SelectionPreview';
export {DownloadButton} from './DownloadButton';
export {BulkDeleter} from './BulkDeleter';
export {AlertModal} from './AlertModal';
export {AlertsDropdown} from './AlertsDropdown';
export {DurationChart} from './DurationChart';
export {ReportTemplateModal, DashboardTemplateModal} from './TemplateModal';
export {KpiCreationModal} from './KpiCreationModal';
export {EmptyState} from './EmptyState';
export {PageTitle} from './PageTitle';
export {TextEditor} from './TextEditor';
export {EntityDescription} from './EntityDescription';
export {Popover} from './Popover';
export {InstanceViewTable} from './InstanceViewTable';
export {TenantInfo} from './TenantInfo';
export {TargetSelection} from './TargetSelection';
export {Loading} from './Loading';
