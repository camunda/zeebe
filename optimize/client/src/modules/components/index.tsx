/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

export {Button} from './Button';
export {Tabs} from './Tabs';
export {Icon} from './Icon';
export {ActionItem} from './ActionItem';
export {Input} from './Input';
export {LabeledInput} from './LabeledInput';
export {Labeled} from './Labeled';
export {Tooltip} from './Tooltip';
export {Modal} from './Modal';
export {Dropdown} from './Dropdown';
export {Select} from './Select';
export {TextEditor} from './TextEditor';
export {PageTitle} from './PageTitle';
export {EntityDescription} from './EntityDescription';
export {EntityNameForm} from './EntityNameForm';
export {LoadingIndicator} from './LoadingIndicator';
export {DocsLink} from './DocsLink';
export {UserTypeahead} from './UserTypeahead';
export {BPMNDiagram, ClickBehavior} from './BPMNDiagram';
export {DMNDiagram} from './DMNDiagram';
export {DownloadButton} from './DownloadButton';
export {NoDataNotice} from './ReportRenderer';
export {Table} from './Table';
export {TargetValueBadge} from './TargetValueBadge';
export {DatePicker} from './DatePicker';
export {DateRangeInput} from './DateRangeInput';
export {Checklist} from './Checklist';
export {DurationChart} from './DurationChart';
export {ColorPicker} from './ColorPicker';
export {Popover} from './Popover';
export {TenantInfo} from './TenantInfo';
export {VersionPopover, TenantPopover} from './DefinitionSelection';
export {CopyToClipboard} from './CopyToClipboard';
export {ShareEntity} from './ShareEntity';
export {ErrorPage} from './ErrorPage';
export {ErrorBoundary} from './ErrorBoundary';
export {TargetSelection} from './TargetSelection';
// @ts-ignore
export {DefinitionSelection} from './DefinitionSelection';
// @ts-ignore
export {ReportRenderer} from './ReportRenderer';
export {Loading} from './Loading';
export {EntityList} from './EntityList';
export {EmptyState} from './EmptyState';

export type {InputProps} from './Input';
export type {LabeledProps} from './Labeled';
export type {User} from './UserTypeahead';
export type {Canvas, ModdleElement, RegistryElement} from './BPMNDiagram';
export type {TooltipProps} from './Tooltip';
export type {TableBody, TableHead} from './Table';
