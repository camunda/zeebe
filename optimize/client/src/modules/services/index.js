/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

export {
  getFlowNodeNames,
  checkDeleteConflict,
  loadDecisionDefinitionXml,
  loadInputVariables,
  loadOutputVariables,
  loadProcessDefinitionXml,
  loadVariables,
} from './dataLoaders';

export {numberParser} from './NumberParser';
export {incompatibleFilters} from './incompatibleFilters.ts';
export {default as reportConfig, createReportUpdate, getDefaultSorting} from './reportConfig';
export {getDiagramElementsBetween} from './diagramServices';
export {default as getTooltipText} from './getTooltipText';
export {default as getScreenBounds} from './getScreenBounds';
export {loadEntity, loadReports, updateEntity, deleteEntity, copyReport} from './entityService';
export {UNAUTHORIZED_TENANT_ID} from './tenantService.ts';
export {loadEntities, copyEntity, createEntity, getEntityIcon} from './entityService.tsx';

export {
  evaluateReport,
  isDurationReport,
  loadRawData,
  getReportResult,
  processResult,
  isAlertCompatibleReport,
  isCategoricalBar,
  isCategorical,
  TEXT_REPORT_MAX_CHARACTERS,
  isTextTileTooLong,
  isTextTileValid,
} from './reportService';

export {addSources, getCollection} from './collectionService';

export * as formatters from './formatters';

export function getRandomId() {
  return Math.random().toString(36).slice(2);
}

export function capitalize(string) {
  return string.charAt(0).toUpperCase() + string.slice(1);
}

export {loadAlerts, addAlert, removeAlert, editAlert} from './alertService';
export {default as ignoreFragments} from './ignoreFragments';
export {default as isReactElement} from './isReactElement';
export {loadDefinitions} from './loadDefinitions';
