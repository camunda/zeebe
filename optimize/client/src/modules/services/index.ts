/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

export function getRandomId() {
  return Math.random().toString(36).slice(2);
}

export {loadReports, loadEntities, copyEntity, createEntity, getEntityIcon} from './entityService';
export {UNAUTHORIZED_TENANT_ID} from './tenantService';
export * as formatters from './formatters';
export {
  loadProcessDefinitionXml,
  loadDecisionDefinitionXml,
  loadInputVariables,
  loadOutputVariables,
  loadVariables,
} from './dataLoaders';
export {numberParser} from './NumberParser';
export {
  TEXT_REPORT_MAX_CHARACTERS,
  isTextTileTooLong,
  isTextTileValid,
  loadRawData,
  evaluateReport,
} from './reportService';
export {addSources, getCollection} from './collectionService';
export {default as getScreenBounds} from './getScreenBounds';
export {default as ignoreFragments} from './ignoreFragments';
export {default as isReactElement} from './isReactElement';
export {incompatibleFilters} from './incompatibleFilters';
export {loadDefinitions} from './loadDefinitions';

export type {Definition} from './loadDefinitions';
export type {ReportEvaluationPayload} from './reportService';
