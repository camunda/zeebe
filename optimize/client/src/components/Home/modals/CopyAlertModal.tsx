/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {useState} from 'react';
import {Button, Form, TextInput} from '@carbon/react';

import {Modal} from 'components';
import {t} from 'translation';

interface CopyAlertModalProps {
  initialAlertName: string;
  onClose: () => void;
  onConfirm: (name: string) => void;
}

export default function CopyAlertModal({
  onClose,
  initialAlertName,
  onConfirm,
}: CopyAlertModalProps) {
  const [name, setName] = useState(initialAlertName + ' (' + t('common.copyLabel') + ')');

  return (
    <Modal open onClose={onClose}>
      <Modal.Header title={t('common.copyName', {name: initialAlertName})} />
      <Modal.Content>
        <Form>
          <TextInput
            id="copyAlertNameInput"
            labelText={t('home.copy.inputLabel')}
            value={name}
            autoComplete="off"
            onChange={({target: {value}}) => setName(value)}
            data-modal-primary-focus
          />
        </Form>
      </Modal.Content>
      <Modal.Footer>
        <Button kind="secondary" onClick={onClose}>
          {t('common.cancel')}
        </Button>
        <Button disabled={!name} onClick={() => onConfirm(name)}>
          {t('common.copy')}
        </Button>
      </Modal.Footer>
    </Modal>
  );
}
