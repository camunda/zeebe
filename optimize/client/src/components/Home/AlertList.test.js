/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import React from 'react';
import {shallow} from 'enzyme';

import {EntityList, AlertModal} from 'components';
import {loadReports, loadAlerts, addAlert} from 'services';

import AlertListWithErrorHandling from './AlertList';
import {getWebhooks} from 'config';
import CopyAlertModal from './modals/CopyAlertModal';

const AlertList = AlertListWithErrorHandling.WrappedComponent;

jest.mock('services', () => {
  const rest = jest.requireActual('services');
  return {
    ...rest,
    loadReports: jest.fn().mockReturnValue([
      {
        id: '1',
        data: {
          visualization: 'table',
          view: {properties: ['frequency']},
          configuration: {aggregationTypes: [{type: 'avg', value: null}]},
        },
        name: 'Report 1',
      },
      {
        id: '2',
        data: {
          visualization: 'number',
          view: {properties: ['duration']},
          configuration: {aggregationTypes: [{type: 'avg', value: null}]},
        },
        name: 'Report 2',
      },
      {combined: true, id: '3', data: {visualization: 'number'}, name: 'Report 3'},
    ]),
    loadAlerts: jest.fn().mockReturnValue([
      {
        id: 'alertID',
        emails: ['test@hotmail.com'],
        name: 'Some Alert',
        lastModifier: 'Admin',
        lastModified: '2017-11-11T11:11:11.1111+0200',
        reportId: '2',
        webhook: null,
      },
    ]),
    addAlert: jest.fn(),
    editAlert: jest.fn(),
    removeAlert: jest.fn(),
  };
});

jest.mock('config', () => ({getWebhooks: jest.fn().mockReturnValue(['webhook1', 'webhook2'])}));

const props = {
  mightFail: jest.fn().mockImplementation((data, cb) => cb(data)),
};

it('should load existing reports', () => {
  shallow(<AlertList {...props} />);

  expect(loadReports).toHaveBeenCalled();
});

it('should load existing webhooks', () => {
  shallow(<AlertList {...props} />);

  expect(getWebhooks).toHaveBeenCalled();
});

it('should only save single number reports', async () => {
  const node = shallow(<AlertList {...props} />);

  expect(node.state('reports').map((report) => report.id)).toEqual(['2']);
});

it('should not show multi-measure reports', async () => {
  loadAlerts.mockReturnValueOnce([]);
  loadReports.mockReturnValueOnce([
    {
      id: '1',
      data: {
        visualization: 'number',
        view: {properties: ['frequency', 'duration']},
        configuration: {aggregationTypes: [{type: 'avg', value: null}]},
      },
      name: 'Report 1',
    },
    {
      id: '2',
      data: {
        visualization: 'number',
        view: {properties: ['duration']},
        configuration: {
          aggregationTypes: [
            {type: 'avg', value: null},
            {type: 'max', value: null},
          ],
        },
      },
      name: 'Report 2',
    },
    {
      id: '3',
      data: {
        visualization: 'number',
        view: {properties: ['frequency']},
        configuration: {
          aggregationTypes: [
            {type: 'avg', value: null},
            {type: 'max', value: null},
          ],
        },
      },
      name: 'Report 3',
    },
  ]);
  const node = shallow(<AlertList {...props} />);

  expect(node.state('reports').map((report) => report.id)).toEqual(['3']);
});

it('should format durations with value and unit', async () => {
  loadAlerts.mockReturnValueOnce([
    {
      id: 'a1',
      name: 'New Alert',
      emails: ['test@hotmail.com'],
      reportId: '2',
      thresholdOperator: '>',
      threshold: {
        value: '12',
        unit: 'seconds',
      },
      checkInterval: {
        value: '10',
        unit: 'minutes',
      },
      reminder: null,
      fixNotification: false,
    },
  ]);

  const node = shallow(<AlertList {...props} />);

  expect(node.find(EntityList).prop('rows')[0].meta[1]).toContain('12s');
});

it('should set the loading', () => {
  const node = shallow(<AlertList {...props} />);

  node.setState({alerts: null}); // simulate missing response from alert query

  expect(node.find(EntityList).prop('isLoading')).toBe(true);
});

it('should load data', () => {
  shallow(<AlertList {...props} />);

  expect(loadReports).toHaveBeenCalled();
});

it('should show information about alerts', async () => {
  const node = shallow(<AlertList {...props} />);

  expect(node.find(EntityList).prop('rows')[0].name).toBe('Some Alert');
  expect(node.find(EntityList).prop('rows')[0].meta[0]).toBe('Report 2');
});

it('should show create Alert button', () => {
  const node = shallow(<AlertList {...props} />);

  expect(node.find(EntityList).prop('action')).toMatchSnapshot();
});

it('should Alert to Deleter', async () => {
  const node = shallow(<AlertList {...props} />);

  node.find(EntityList).prop('rows')[0].actions[2].action();

  expect(node.state('deleting')).toEqual({
    id: 'alertID',
    emails: ['test@hotmail.com'],
    name: 'Some Alert',
    lastModifier: 'Admin',
    lastModified: '2017-11-11T11:11:11.1111+0200',
    reportId: '2',
    webhook: null,
  });
});

it('should open a modal when editing an alert', async () => {
  const node = shallow(<AlertList {...props} />);

  node.find(EntityList).prop('rows')[0].actions[0].action();

  expect(node.find(AlertModal)).toExist();
  expect(node.find(AlertModal).prop('initialAlert')).toEqual({
    id: 'alertID',
    emails: ['test@hotmail.com'],
    name: 'Some Alert',
    lastModifier: 'Admin',
    lastModified: '2017-11-11T11:11:11.1111+0200',
    reportId: '2',
    webhook: null,
  });
});

it('should invoke addAlert when copying an alert', async () => {
  const node = shallow(<AlertList {...props} />);

  node.find(EntityList).prop('rows')[0].actions[1].action();

  node.find(CopyAlertModal).prop('onConfirm')('testName');

  expect(addAlert).toHaveBeenCalledWith({
    id: 'alertID',
    emails: ['test@hotmail.com'],
    name: 'testName',
    lastModifier: 'Admin',
    lastModified: '2017-11-11T11:11:11.1111+0200',
    reportId: '2',
    webhook: null,
  });
});

it('should show warning if alert is inactive due to missing webhoook', () => {
  getWebhooks.mockReturnValueOnce(['webhook1']);
  loadAlerts.mockReturnValueOnce([
    {
      id: 'alertID',
      emails: [],
      name: 'Some Alert',
      lastModifier: 'Admin',
      lastModified: '2017-11-11T11:11:11.1111+0200',
      reportId: '2',
      webhook: 'nonExistingWebhook',
    },
  ]);
  const node = shallow(<AlertList {...props} />);

  expect(node.find(EntityList).prop('rows')[0].warning).toBe('Alert inactive');
});
