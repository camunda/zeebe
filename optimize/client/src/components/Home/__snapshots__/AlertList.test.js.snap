// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show create Alert button 1`] = `
<Button
  className="createAlert"
  kind="primary"
  onClick={[Function]}
>
  Create new alert
</Button>
`;
